modules/layers/src/solid-polygon-layer/polygon-tesselator.js

```diff
@@ -70,9 +70,9 @@ export default class PolygonTesselator extends Tesselator {
           target,
           size: 4,
           getValue: object => {
-            const color = accessor(object);
-            if (isNaN(color[3])) {
-              color[3] = 255;
+            let color = accessor(object);
+            if (color.length < 4) {
+              color = [...color, 255];
```

Comment:
This can be very expensive. Can you create an intermediate target e.g.

```
const colorArray = [0, 0, 0, 255];
return this._updateAttribute({
  ...
  getValue: object => {
    const color = accessor(object);
    if (color.length === 4} {
      return color;
    }
    colorArray[0] = color[0];
    colorArray[1] = color[1];
    colorArray[2] = color[2];
    return colorArray;
  }
```

Reply:
@Pessimistress how would this ensure that the color values are correctly copied to the buffer before `colorArray` is mutated by the next call of the accessor?

Reply:
@manassra `_updateAttributes` immediately copies the result of `getValue` into the attribute array: https://github.com/uber/deck.gl/blob/b520d5d91324780f9604ff801b9085af6da32e26/modules/core/src/utils/tesselator.js#L126-L129

Further, I'd recommend that the `colorArray` that @Pessimistress suggested be pulled out of the class completely so it's only ever allocated once.

Reply:
> how would this ensure that the color values are correctly copied to the buffer before colorArray is mutated by the next call of the accessor?

The color object is used once to populate a chunk in the attribute typed array. It is no longer needed after the current object is processed.

Reply:
@Pessimistress @tsherif thank you for confirming! I just wanted to make sure the value doesn't get overwritten before it is actually used.